Header header
bool wall_left
bool wall_right
bool wall_front